functions/api/category.js

```diff
@@ -0,0 +1,386 @@
+const constants = require('../config/constants')
+const validate = require('../validation/validation')
+const logger = require('../middleware/logger');
+const config = require('config');
+const joi = require('@hapi/joi');
+const admin = require('firebase-admin');
+const functions = require('firebase-functions');
+const express = require('express');
+const router = express.Router();
+const cors = require('cors');
+const db = admin.firestore();
+
+/**
+ * @description Route to retireve all categories from firestore
+ * @returns Json object containing all categories
+ */
+router.get("/", async (request, response) => {
+    logger.info("Retrieving all categories from firestore");
+    const categories = {
+        "categories": []
+    }
+    let categoryCollection = db.collection(constants.CATEGORY);
+    let snapshot = await categoryCollection.get()
+    snapshot.forEach(category => {
+        var categoryInfo = {}
+        var categoryData = category.data()
+        categoryInfo[constants.CATEGORY] = category.id,
```

Comment:
any reason why we are assigning every attribute to another object. Why cant we just do categories.categories.push(category);

Same comment for all the other entities and apis